recipe/activate.sh

```diff
@@ -15,6 +15,8 @@ if [ "${CONDA_BUILD:-0}" = "1" ]; then
         exit 1
     fi
     CUDA_INCLUDE_DIR="${PREFIX}/${targetsDir}/include"
+    # Needed to fix cross compilation
+    export CMAKE_ARGS="${CMAKE_ARGS} -DCMAKE_FIND_ROOT_PATH=$PREFIX;$BUILD_PREFIX/$HOST/sysroot;$BUILD_PREFIX/${targetsDir}"
```

Comment:
So for cross compiling `CONDA_PREFIX` maps to `BUILD_PREFIX`? Because if it maps to `$PREFIX` that will already be added as a find root path.

CMake currently adds `$BUILD_PREFIX/$HOST/sysroot/usr` as a search location ( via `gcc --print-sysroot` ), what does conda have in `$BUILD_PREFIX/$HOST/sysroot` that isn't in `$BUILD_PREFIX/$HOST/sysroot/usr`?

Reply:
Basically `build` & `host` are 2 different Conda environments created based on `requirements`

`$PREFIX` is the path to the `host` environment

`$BUILD_PREFIX` is the path to the `build` environment

Reply:
`$CONDA_PREFIX` is effectively equivalent to `$PREFIX` (just in the end user's activated environment)

Reply:
`CONDA_PREFIX` maps to `BUILD_PREFIX`.

> CMake currently adds $BUILD_PREFIX/$HOST/sysroot/usr as a search location ( via gcc --print-sysroot ),

Yes, `gcc -print-sysroot` gives `$BUILD_PREFIX/$HOST/sysroot/`. Is this cmake behaviour new? I don't recall this being there when we added it.

Reply:
CMake 3.18 introduced the behavior of querying the compiler for `--print-sysroot` and adding paths based on it.

We only add `$BUILD_PREFIX/$HOST/sysroot/usr` for the reasons discussed here ( and the linked issue ).
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/4585#note_729513